app/lib/common/toolkit/errors/util.dart

```diff
@@ -1,12 +1,20 @@
+typedef ErrorTextBuilder = String Function(Object error, ErrorCode code);
+
 enum ErrorCode {
   notFound,
+  forbidden,
```

Comment:
Extended the Error types to be more specific and more precise and _allow us_ to show a different error depending on the underlying code we were given by the server: most notably if we were forbidden access.